### PR TITLE
tilt: m1 flag

### PR DESCRIPTION
**Pull request by evan-gray**
Keeping the m1 logic separate for now to avoid affecting working builds

---
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1202)

### CODE REVIEW

**Comment by evan-gray** [comment]
Tilt now works on m1 with `tilt up -- --m1`

File: solana/Dockerfile.m1 [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    libudev-dev \
+    llvm \
+    pkg-config \
+    zlib1g-dev \
+    && \
+    rm -rf /var/lib/apt/lists/* && \
+    rustup component add rustfmt && \
+    rustup toolchain add --profile minimal nightly-2021-12-03
+
+FROM stage-0 as stage-1
```
Comment (jynnantonix) [inline solana/Dockerfile.m1:30]: Same comment here about removing the stage-1 step.

File: solana/Dockerfile.wasm.m1 [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# syntax=docker.io/docker/dockerfile:1.3@sha256:42399d4635eddd7a9b8a24be879d2f9a930d0ed040a61324cfdf59ef1357b3b2
+FROM docker.io/library/rust:1.49@sha256:a50165ea96983c21832578afb1c8c028674c965bc1ed43b607871b1f362e06a5 AS build
+
+RUN apt-get update && apt-get install -y libssl-dev libudev-dev pkg-config zlib1g-dev llvm clang
+RUN rustup component add rustfmt
+RUN rustup default nightly-2022-01-02
```
Comment (jynnantonix) [inline solana/Dockerfile.wasm.m1:6]: We probably want this to match the toolchain we're using for all other solana code: nightly-2021-12-03

File: Tiltfile [hunk 3]
```diff
@@ -19,6 +19,7 @@ update_settings(max_parallel_updates = 10)
 # Runtime configuration
 config.define_bool("ci", False, "We are running in CI")
 config.define_bool("manual", False, "Set TRIGGER_MODE_MANUAL by default")
+config.define_bool("m1", False, "Build Solana from source for m1 arm64")
```
Comment (jynnantonix) [inline Tiltfile:22]: Naming nit: how about "arm64"?  The build steps will work fine on non-M1 arm64 machines.

**Pull request closed without merging**